src/controllers/productController.js

```diff
@@ -26,48 +26,47 @@ class productController{
     }
 
   static async createProduct(req, res) {
-    try {
-
-      if (req.user.role.roleName !== "vendor") {
-        return res
-          .status(401)
-          .json({ status: "fail", message: req.t("Unauthorized") });
-      };
-
-     const {
-        productName,
-        productPrice,
-        quantity,
-        productDescription,
-        productOwner,
-        CategoryId,
-        expiredDate
-      } = req.body;
-      const productImage = req.files.map((img) => img.path);
-      const product = await Product.create({
-        vendorId: req.user.role.id,
-        productName,
-        productPrice,
-        quantity,
-        productDescription,
-        productOwner,
-        expiredDate,
-        CategoryId,
-        productImage
-      });
-
-      await product.save();
-      return res
-        .status(200)
-        .json({
+    if (req.user.roleName === "vendor") {
+      try {
+        const {
+          productName,
+          productPrice,
+          quantity,
+          productDescription,
+          productOwner,
+          CategoryId,
+          bonus,
+          expiredDate
+        } = req.body;
+        const productImage = req.files.map((img) => img.path);
+        const product = await Product.create({
+          vendorId: req.user.id,
+          productName,
+          productPrice,
+          quantity,
+          productDescription,
+          productOwner,
+          expiredDate,
+          CategoryId,
+          bonus,
+          productImage
+        });
+        await product.save();
```

Comment:
`await product.save();` is not needed since `create` method does it all a once on this other PR review [here](https://github.com/atlp-rwanda/ecommerce-app-champions-bn/pull/26#discussion_r1151400258)